src/main/python/ayab/plugins/ayab_plugin/ayab_control.py

```diff
@@ -181,88 +215,94 @@ def __cnfLine(self, lineNumber):
             # adjust lineNumber with current block
             lineNumber += BLOCK_LENGTH * self.__lineBlock
 
-            imgHeight = self._image.imgHeight()
-        
             # work out which line of data to send
-            imgRow, color, indexToSend, sendBlankLine, lastLine = self._knit_func(lineNumber, imgHeight)
+            imgRow, color, indexToSend, sendBlankLine, lastLine = self.__knit_func(lineNumber)
             
             # create bitarray
             bits = self._select_needles(color, indexToSend, sendBlankLine)
 
-            # TODO implement CRC8
-            crc8 = 0x00
-
             # send line to machine
-            if self._infRepeat:
-                self.__ayabCom.cnf_line(requestedLine, bits.tobytes(), 0, crc8)
-            else:
-                self.__ayabCom.cnf_line(requestedLine, bits.tobytes(), lastLine, crc8)
+            self.__ayabCom.cnf_line(requestedLine, bits.tobytes(), lastLine & self.__infRepeat)
```

Comment:
AyabCommunicationMockup.cnf_line() still expects (lineNumber, lineData, flags, crc8). Please remove the crc8 parameter so we have the same interface for both variants

Reply:
OK